CRM/Contact/Page/View/Relationship.php

```diff
@@ -134,6 +134,10 @@ public function view() {
    */
   public function browse() {
     // do nothing :) we are using datatable for rendering relationship selectors
+    $columnHeaders = CRM_Contact_BAO_Relationship::getColumnHeaders();
+    $contactRelationships = $selector = NULL;
+    CRM_Utils_Hook::searchColumns('relationship.columns', $columnHeaders, $contactRelationships, $selector);
```

Comment:
Why do we need the hook here - in other places it is very centralised - but they do seem to be actually using the selector whereas I think you are mocking it?

Reply:
We are mocking it and there are two different functions - the page loads with the columns here.  Later the datatable loads the rows via an ajax call.